test/extended/images/mysql_replica.go

```diff
@@ -0,0 +1,311 @@
+	defer GinkgoRecover()
+	var (
+		nfsTemplatePath = exutil.FixturePath("..", "..", "test", "extended", "fixtures", "nfs-pv-provider.json")
+		templatePath    = "https://raw.githubusercontent.com/openshift/mysql/master/5.5/examples/replica/mysql_replica.json"
+		//templatePath = "https://raw.githubusercontent.com/openshift/mysql/master/5.6/examples/replica/mysql_replica.json"
+		oc       = exutil.NewCLI("mysql-replication", exutil.KubeConfigPath())
+		user     = "user"
+		userPass = "userpass"
+		rootPass = "rootpass"
+		database = "userdb"
+	)
+
+	// TODO: Rename all out* vars to out
+	Describe("MySQL image replication", func() {
+		JustBeforeEach(func() {
+			// FIXME: Shouldn't do this globaly
+			err := oc.AsAdmin().Run("patch").Args("scc", "privileged", "-p", "{\"users\":[\"system:serviceaccount:openshift-infra:build-controller\",\"system:serviceaccount:default:router\",\"system:serviceaccount:"+oc.Namespace()+":default\"]}").Verbose().Execute()
+			Expect(err).NotTo(HaveOccurred())
+			nfsFile, err := oc.Run("process").Args("-f", nfsTemplatePath, "-v", "NAMESPACE="+oc.Namespace()).Verbose().OutputToFile("nfs.json")
+			Expect(err).NotTo(HaveOccurred())
+
+			By(fmt.Sprintf("calling oc create -f %q", nfsFile))
+			err = oc.AsAdmin().Run("create").Args("-f", nfsFile).Verbose().Execute()
+			Expect(err).NotTo(HaveOccurred())
+
+			err = oc.KubeFramework().WaitForAnEndpoint("nfs")
+			Expect(err).NotTo(HaveOccurred())
+			pods, err := getPods(oc, "name=nfs")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(pods).To(HaveLen(1))
+			err = oc.KubeFramework().WaitForPodRunning(pods[0])
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			err := oc.AsAdmin().Run("delete").Args("pods", "--all").Verbose().Execute()
+			Expect(err).NotTo(HaveOccurred())
+			err = oc.AsAdmin().Run("delete").Args("all", "--all").Verbose().Execute()
+			Expect(err).NotTo(HaveOccurred())
+			err = oc.AsAdmin().Run("delete").Args("pv,pvc", "--all").Verbose().Execute()
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should work", func() {
+			oc.SetOutputDir(exutil.TestContext.OutputDir)
+
+			By(fmt.Sprintf("processing and creating the template %q", templatePath))
+			configFile, err := oc.Run("process").Verbose().
+				Args(
+				"-f", templatePath,
+				"-v", "MYSQL_USER="+user+",MYSQL_PASSWORD="+userPass+",MYSQL_ROOT_PASSWORD="+rootPass+",MYSQL_DATABASE="+database).
+				OutputToFile("config.json")
+			Expect(err).NotTo(HaveOccurred())
+			err = oc.Run("create").Args("-f", configFile).Execute()
+			Expect(err).NotTo(HaveOccurred())
+
+			By("seeing the mysql-master endpoint is up")
+			err = oc.KubeFramework().WaitForAnEndpoint("mysql-master")
+			Expect(err).NotTo(HaveOccurred())
+
+			By("seeing the mysql-slave endpoint is up")
+			err = oc.KubeFramework().WaitForAnEndpoint("mysql-slave")
+			Expect(err).NotTo(HaveOccurred())
+
+			// Get pod names
+			podNames, err := getPods(oc, "name=mysql-master")
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
you will have to prefix **all** Ginkgo/Gomega functions with `g.` and `o.`. See the builds test.